tripleo/utils/types.py

```diff
@@ -13,92 +13,126 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 """
-from abc import ABC
+from abc import ABC, abstractmethod
 from os import PathLike
 from pathlib import Path
 from typing import Any, Dict, Union
 
+from overrides import overrides
+
+from tripleo.utils.paths import Preprocessor
 from tripleo.utils.strings import is_url
 
+# -- Serialization --
 YAML = Dict[str, Any]
 """Represents data originated from reading a YAML file."""
 
+# -- FileSystem --
+RawPath = Union[bytes, str, PathLike, Path]
+
 
-class FSPath(ABC):
+class FSPath(str, ABC):
     """Base class for representations of a filesystem path. These are able
     to make system calls and are meant to model elements existing on the
     filesystem.
     """
 
-    def __init__(self, path: Union[bytes, str, PathLike, Path]):
+    def __new__(cls, value: RawPath, *makeup: Preprocessor):
```

Comment:
I'm struggling to understand how  `makeup` arguments are passed here. Shouldn't the subclassed of `FSPath` somehow modify the constructor to bind them? Is there any example of it being used?

Reply:
There is one in my other PR:
`DEFAULT_CLONE_PATH = Dir('~/.cibyl', resolve_home)`